### PR TITLE
feat(linter): rename @nx/eslint-plugin-nx to @nx/eslint-plugin

### PR DESCRIPTION
**Pull request by FrozenPandaz**
## Current Behavior

The new `@nx/eslint-plugin-nx` package means the rules look like `@nx/nx/enforce-module-boundaries` which is redundant.

## Expected Behavior

The new `@nx/eslint-plugin-nx` package is now `@nx/eslint-plugin` means the rules look like `@nx/enforce-module-boundaries` which is better.

## Related Issue(s)

Fixes #

### CODE REVIEW

**Review by AgentEnder — CHANGES_REQUESTED** [review]
Can you update codeowners for the eslint-plugin-nx directory?

**Review by barbados-clemens — APPROVED** [review]

**Review by nartc — APPROVED** [review]

**Review by Coly010 — APPROVED** [review]

**Review by mandarini — APPROVED** [review]

File: packages/eslint-plugin/src/constants.ts
Comment (JamesHenry) [inline packages/eslint-plugin/src/constants.ts:1]: There is still a reference to `eslint-plugin-nx` in the comment here, please be sure to do a global find for `eslint-plugin-nx` and other relevant strings

**Review by JamesHenry — CHANGES_REQUESTED** [review]

File: docs/generated/manifests/packages.json [hunk 1]
```diff
@@ -645,40 +645,40 @@
     },
     "path": "/packages/esbuild"
   },
-  "eslint-plugin-nx": {
+  "eslint-plugin": {
     "githubRoot": "https://github.com/nrwl/nx/blob/master",
-    "name": "eslint-plugin-nx",
-    "packageName": "@nrwl/eslint-plugin-nx",
```
Comment (meeroslav) [inline docs/generated/manifests/packages.json:651]: This should stay the same until we publish it - `@nrwl/eslint-plugin-nx`

File: docs/generated/packages-metadata.json [hunk 2]
```diff
@@ -639,38 +639,38 @@
...
     ],
     "executors": [],
     "generators": [],
     "githubRoot": "https://github.com/nrwl/nx/blob/master",
-    "name": "eslint-plugin-nx",
-    "packageName": "@nrwl/eslint-plugin-nx",
-    "root": "/packages/eslint-plugin-nx",
-    "source": "/packages/eslint-plugin-nx/src"
+    "name": "eslint-plugin",
+    "packageName": "@nrwl/eslint-plugin",
```
Comment (meeroslav) [inline docs/generated/packages-metadata.json:671]: This is invalid - should be `@nrwl/eslint-plugin-nx`

**Review by meeroslav — CHANGES_REQUESTED** [review]
Left few minor comments

File: packages-legacy/linter/project.json [hunk 3]
```diff
@@ -32,6 +32,5 @@
         ]
       }
     }
-  },
-  "implicitDependencies": ["eslint-plugin-nx"]
```
[the remark this answers is not in the document]
Comment (FrozenPandaz) [reply packages-legacy/linter/project.json:36]: This is the legacy package, it only has a dependency on `@nx/linter`

File: docs/generated/manifests/packages.json (hunk 1 above)
Comment (FrozenPandaz) [reply docs/generated/manifests/packages.json:651]: Fixed :)

File: packages/eslint-plugin/src/constants.ts
Comment (FrozenPandaz) [reply packages/eslint-plugin/src/constants.ts:1]: Fixed :)

File: packages-legacy/linter/project.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (meeroslav) [reply packages-legacy/linter/project.json:36]: Missed that suffix. Tnx

**Review by meeroslav — APPROVED** [review]
Looks good now, thanks!

**Review by jaysoo — APPROVED** [review]

File: packages/eslint-plugin/jest.config.ts [hunk 4]
```diff
@@ -5,7 +5,7 @@ export default {
   },
   moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
   globals: {},
-  displayName: 'eslint-plugin-nx',
+  displayName: 'packages-eslint-plugin',
```
Comment (AgentEnder) [inline packages/eslint-plugin/jest.config.ts:8]: non-blocking: Does this need the `packages-` prefix? We don't use that elsewhere for this

**Review by AgentEnder — APPROVED** [review]

**Pull request merged**